packages/core/scripts/setup/ios.ts

```diff
@@ -4,92 +4,78 @@ const path = require('path');
-          return false;
-        }
-        appDelegate.addAfter('#import "AppDelegate.h"', embraceImport);
-        appDelegate.patch();
-        return true;
-      }),
-  docURL:
-    'https://embrace.io/docs/react-native/integration/add-embrace-sdk/?platform=ios#manually',
+export const tryToPatchAppDelegate = async ({
+  name,
+}: {
+  name: string;
+}): Promise<boolean> => {
+  const appDelegate = getAppDelegateByIOSLanguage(name, 'objectivec');
+  if (!appDelegate) {
+    const appDelegateSwift = getAppDelegateByIOSLanguage(name, 'swift');
+    if (!appDelegateSwift) {
+      logger.format(
+        'Couldn\'t find AppDelegate. Please refer to the docs at https://embrace.io/docs/react-native/integration/add-embrace-sdk/?rn-platform=ios&platform=ios to update manually.'
+      );
+      return false;
+    }
+    return await patchAppDelegateSwift(appDelegateSwift);
+  }
+  return (await patchAppDelegateObjectiveC(appDelegate)) || false;
 };
 
 export const iosInitializeEmbrace = {
   name: 'iOS initialize Embrace',
   run: (wizard: Wizard): Promise<any> =>
-    wizard
-      .fieldValue(packageJSON)
-      .then((json) => appDelegatePatchable(json))
-      .then((appDelegate) => {
-        logger.log('patching AppDelegate with Embrace initialize');
-        if (appDelegate.hasLine('[[Embrace sharedInstance] start')) {
-          logger.warn('already initializing Embrace');
-          return false;
-        }
-        const embraceInitializer = formatEmbraceInitializer();
-        const DID_LAUNCH_REGEX =
-          /(-\s*\(BOOL\)\s*application:\s*\(UIApplication\s\*\)\s*(app|application)\s+didFinishLaunchingWithOptions:\s*\(NSDictionary\s*\*\)launchOptions\s*\{\s*)/;
-
-        appDelegate.addAfter(DID_LAUNCH_REGEX, embraceInitializer);
-        appDelegate.patch();
-        return true;
-      }),
+    wizard.fieldValue(packageJSON).then(tryToPatchAppDelegate),
   docURL:
     'https://embrace.io/docs/react-native/integration/add-embrace-sdk/?platform=ios#manually',
 };
 
+export const patchPodfile = (json: IPackageJson) => {
+  const rnVersion = (json.dependencies || {})['react-native'];
+  if (!rnVersion) {
+    throw Error('react-native dependency was not found');
+  }
+  const rnVersionSanitized = rnVersion.replace('^', '');
+
+  // If 6.0.0, autolink should have linked the Pod.
+  if (semverGte('6.0.0', rnVersionSanitized)) {
+    logger.log(
+      'skipping patching Podfile since react-native is on an autolink supported version'
+    );
+    return;
+  }
+  return podfilePatchable().then((podfile) => {
+    if (podfile.hasLine(embraceNativePod)) {
+      logger.warn('Already has EmbraceIO pod');
+      return;
+    }
+    podfile.addBefore('use_react_native', `${embraceNativePod}\n`);
+
+    return podfile.patch();
+  });
+};
+
 export const iosPodfile = {
   name: 'Podfile patch (ONLY React Native Version < 0.6)',
   run: (wizard: Wizard): Promise<any> =>
-    wizard.fieldValue(packageJSON).then((json) => {
```

Comment:
what was the reason for moving this above?

Reply:
Unit test